src/gossip/data.zig

```diff
@@ -1084,26 +1087,30 @@ pub const ContactInfo = struct {
         try sanitizeWallclock(self.wallclock);
     }
 
-    pub fn getSocket(self: *const Self, key: u8) ?SocketAddr {
-        if (self.cache[key].eql(&SocketAddr.UNSPECIFIED)) {
+    pub fn getSocket(self: *const Self, key: socket_tag) ?SocketAddr {
+        const maybe_socket = &self.cache[@intFromEnum(key)];
```

Comment:
this should be `socket` since its not an optional type

Reply:
2e1acea6eaf8481ceae012a049655822a2aabda4

Reply:
nice ty - cant inspect the commit changes exactly (bc of the force push i think - extra commit history after reviews via not force pushing is ok imo)